pkg/security/x509.go

```diff
@@ -51,8 +54,9 @@ func newTemplate(commonName string, lifetime time.Duration) (*x509.Certificate,
 		return nil, err
 	}
 
-	notBefore := timeutil.Now().Add(validFrom)
-	notAfter := notBefore.Add(lifetime)
+	now := timeutil.Now()
```

Comment:
the extra padding (see reason on L40) meant that users weren't getting the lifetime they expected. I switched to using the actual expiration date and padding the "NotBefore" date.